java.lang.OutOfMemoryError: Java heap space when OpenAPI is enabled and enums are used in a JAX-RS endpoint

**Describe the bug**
An OOME is thrown when the `quarkus-smallrye-openapi` extension is used and a JAX-RS endpoint has an enum as a query parameter.
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:0.23.2:build (default) on project getting-started: Failed to build a runnable JAR: Failed to build a runner jar: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build threw an exception: java.lang.OutOfMemoryError: Java heap space
[ERROR] 	at java.util.Arrays.copyOfRange(Arrays.java:3664)
[ERROR] 	at java.lang.String.<init>(String.java:207)
[ERROR] 	at java.lang.StringBuilder.toString(StringBuilder.java:407)
[ERROR] 	at com.fasterxml.jackson.core.util.TextBuffer.contentsAsString(TextBuffer.java:404)
[ERROR] 	at com.fasterxml.jackson.core.io.SegmentedStringWriter.getAndClear(SegmentedStringWriter.java:83)
[ERROR] 	at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString(ObjectWriter.java:999)
[ERROR] 	at io.smallrye.openapi.runtime.io.OpenApiSerializer.serialize(OpenApiSerializer.java:107)
[ERROR] 	at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.build(SmallRyeOpenApiProcessor.java:229)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:780)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
[ERROR] 	at io.quarkus.builder.BuildContext$$Lambda$101/958922413.run(Unknown Source)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR] 	at java.lang.Thread.run(Thread.java:748)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]

```

**Expected behavior**
No error

**Actual behavior**
OutOfMemoryError during build

**To Reproduce**
Steps to reproduce the behavior:
1. Create a project from code.quarkus.io and choose the `quarkus-smallrye-openapi` extension
2. Create an enum and add it as a parameter in the sample endpoint
3. mvn clean package and you should see the error

**Configuration**
```properties
# Add your application.properties here, if applicable.

```
**Environment:**
 - Output of `uname -a` or `ver`: `Linux ggastald-laptop 5.2.15-200.fc30.x86_64 #1 SMP Mon Sep 16 15:17:36 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`:  
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b03)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b03, mixed mode)
```
 - GraalVM version (if different from Java): Not used
 - Quarkus version or git rev: 0.23.2

**Additional context**

It works in Quarkus 0.22.0

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -34,7 +34,7 @@
         <smallrye-config.version>1.3.9</smallrye-config.version>
         <smallrye-health.version>2.0.0</smallrye-health.version>
         <smallrye-metrics.version>2.1.4</smallrye-metrics.version>
-        <smallrye-open-api.version>1.1.8</smallrye-open-api.version>
+        <smallrye-open-api.version>1.1.9</smallrye-open-api.version>
         <smallrye-opentracing.version>1.3.2</smallrye-opentracing.version>
         <smallrye-fault-tolerance.version>2.0.9</smallrye-fault-tolerance.version>
         <smallrye-jwt.version>2.0.9</smallrye-jwt.version>

--- a/extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/OpenApiResource.java
+++ b/extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/OpenApiResource.java
@@ -2,6 +2,7 @@ package io.quarkus.smallrye.openapi.test;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
 
 @Path("/resource")
 public class OpenApiResource {
@@ -9,5 +10,16 @@ public class OpenApiResource {
     @GET
     public String root() {
         return "resource";
+    }
+
+    @GET
+    @Path("/test-enums")
+    public Query testEnums(@QueryParam("query") Query query) {
+        return query;
+    }
+
+    public enum Query {
+        QUERY_PARAM_1,
+        QUERY_PARAM_2;
     }
 }

